pkg/datasource.go

```diff
@@ -798,12 +805,16 @@ func (o *OCIDatasource) processLogMetrics(ctx context.Context, searchLogsReq Gra
 					for rowCount, logSearchResult := range res.SearchResponse.Results {
 						searchResultData, ok := (*logSearchResult.Data).(map[string]interface{})
 						if ok == true {
-							// If this is the first row then inspect the values of the elements to
-							// speed up the processing of the remaining rows
-							if rowCount == 0 {
+							// If this is the first row for the first interval then inspect the
+							// values of the elements to speed up the processing of the remaining rows
+							// for all intervals. It is important to do this only for the first row of
+							// all of the results otherwise the order of the label keys may be different
+							// between the search results for different intervals
+							if intervalCnt == 0 && rowCount == 0 {
 								// Loop through the keys for the entries in the results data item
 								// to determine what kind of fields we have in the results
 								for key, value := range searchResultData {
+
 									// Check whether the key contains one of the aggregation functions
 									if key == "count" {
 										metricFieldName = "count"
```

Comment:
I don't think you changed this line but saw that metricFieldName is always value(key) whether the value is "count" or something in both if and else clause. Can we not pull this assignment outside??

Reply:
I don't think so since in the case that the query has a group by clause with one or more other fields then these are considered label fields in the Grafana data frame vernacular and will be different keys in the results, e.g.

```
{
  "results" : [ {
    "data" : {
      "data.namespace" : "cne-cred-manager",
      "sum(data.lsbeat_eventSize)" : 1232661
    }
  }, {
    "data" : {
      "data.namespace" : "kube-system",
      "sum(data.lsbeat_eventSize)" : 11120233
    }
  },
```
So if the key is "data.namespace" we don't want to set the `metricFieldName` variable.

Reply:
Probably the if and else if clauses could be combined together